build_resolvers/test/resolver_test.dart

```diff
@@ -374,6 +374,22 @@ int? get x => 1;
               }, resolvers: AnalyzerResolvers()),
           ['non-nullable']);
     }, skip: 'Requires the next version of analyzer which isnt yet published.');
+
+    group('regressions', () {
+      test('can use analysis session after resolving additional assets',
+          () async {
+        var resolvers = AnalyzerResolvers();
+        await resolveSources({
+          'a|web/main.dart': '',
+          'a|web/other.dart': '',
+        }, (resolver) async {
+          var lib = await resolver.libraryFor(entryPoint);
+          expect(
+              await resolver.isLibrary(AssetId('a', 'web/other.dart')), true);
+          expect(lib.session.getResolvedLibraryByElement(lib), isEmpty);
```

Comment:
Is `isEmpty` the right matcher? How about `completes`?

Reply:
I went with an explicit `await` and `isNotNull`.